core/src/main/java/org/commonjava/indy/core/conf/DefaultIndyConfigFactory.java

```diff
@@ -129,22 +128,20 @@ public synchronized void load( final String configPath )
         InputStream stream = null;
         try
         {
-            stream = ConfigFileUtils.readFileWithIncludes( config, props );
-
-            new DotConfConfigurationReader( listeners ).loadConfiguration( stream );
-
-            Properties sysprops = System.getProperties();
-            props.stringPropertyNames().forEach( ( name ) -> sysprops.setProperty( name, props.getProperty( name ) ) );
+            Properties props = getBaseSystemProperties();
 
             configSections.forEach( (section)->{
                 if ( section instanceof SystemPropertyProvider)
                 {
                     Properties p = ( (SystemPropertyProvider) section ).getSystemProperties();
-                    p.stringPropertyNames().forEach( ( name ) -> sysprops.setProperty( name, p.getProperty( name ) ) );
+                    p.stringPropertyNames().forEach( ( name ) -> props.setProperty( name, p.getProperty( name ) ) );
                 }
             });
 
-            System.setProperties( sysprops );
+            stream = ConfigFileUtils.readFileWithIncludes( config );
+            new DotConfConfigurationReader( listeners ).loadConfiguration( stream, props );
+
+            System.setProperties( props );
```

Comment:
*NOTE:* The environment variables (prefixed with `env.`) are INCLUDED in the System properties after this point. This will make them available for use when we resolve expressions in the infinispan.xml, for example, as long as that happens AFTER this load method is called.